### PR TITLE
Minor updates to ExprTk

### PR DESCRIPTION
**Pull request by ArashPartow**
- Updated unknown symbol resolver interface to handle all types (scalar, string and vector)
- Added compile-time check for vector indexing when using constant values
- Added return statement enable/disable via parser settings
- Added exprtk_disable_return_statement macro for disabling return statements and associated exceptions at the source code level.

### CODE REVIEW

File: libs/base/include/mrpt/otherlibs/exprtk.hpp [hunk 1]
```diff
@@ -2,14 +2,14 @@
...
+ * Author: Arash Partow (1999-2017)                               *
  * URL: http://www.partow.net/programming/exprtk/index.html       *
  *                                                                *
  * Copyright notice:                                              *
  * Free use of the C++ Mathematical Expression Toolkit Library is *
  * permitted under the guidelines and in accordance with the most *
- * current version of the Common Public License.                  *
- * http://www.opensource.org/licenses/cpl1.0.php                  *
+ * current version of the MIT License.                            *
+ * http://www.opensource.org/licenses/MIT                         *
```
Comment (jolting) [inline libs/base/include/mrpt/otherlibs/exprtk.hpp]: Good call. That indemnification clause is trouble.

Comment (ArashPartow) [inline libs/base/include/mrpt/otherlibs/exprtk.hpp]: Makes it easier for more people to use without having to get lawyers involved.  :D

Comment (jolting) [inline libs/base/include/mrpt/otherlibs/exprtk.hpp]: Can you fix it here too? https://github.com/MRPT/mrpt/blob/89415f08c6086dc7d726ce767706a1ce2ff33093/packaging/debian/copyright#L771

Comment (jolting) [inline libs/base/include/mrpt/otherlibs/exprtk.hpp]: Woops. a copyright.bak got introduced

Comment (ArashPartow) [inline libs/base/include/mrpt/otherlibs/exprtk.hpp]: done.

Comment (ArashPartow) [inline libs/base/include/mrpt/otherlibs/exprtk.hpp]: Sorry about that, it's really late here.  It's fixed now.

File: libs/base/include/mrpt/otherlibs/exprtk.hpp [hunk 2]
```diff
@@ -26248,22 +26571,30 @@ namespace exprtk
...
-               case_stmt(details::e_avg,  details::vararg_avg_op  )
-               case_stmt(details::e_min,  details::vararg_min_op  )
-               case_stmt(details::e_max,  details::vararg_max_op  )
-               case_stmt(details::e_mand, details::vararg_mand_op )
-               case_stmt(details::e_mor,  details::vararg_mor_op  )
-               case_stmt(details::e_multi,details::vararg_multi_op)
-               #undef case_stmt
-               default : return error_node();
+               return node_allocator_->
+                 allocate<details::str_vararg_node<Type,details::vararg_multi_op<Type> > >(arg_list);
```
Comment (jolting) [inline libs/base/include/mrpt/otherlibs/exprtk.hpp]: str_vararg_node is not in the details namespace.

Comment (ArashPartow) [inline libs/base/include/mrpt/otherlibs/exprtk.hpp]: It's in the namespace, but in MRPT string functionality is disabled, and this is a new node I'd added recently and hadn't tested it properly with strings disabled. hence the compile problem. I've fixed the code that invokes it.

**Pull request merged**